### PR TITLE
Changing defaults in mitochondria mode for M2 filtering rehaul

### PR DESCRIPTION
**Pull request by meganshand**
After #5688 we need to adjust some of the default values when in mitochondria mode. I tested these parameters with the mixture samples and it looked good. In particular note that the TLOD divided by depth filter is no longer needed, so the default would now be 0.

Again, would love to get this in before the release on Tuesday (fyi @droazen)

@ldgauthier @davidbenjamin @jsotobroad

### CODE REVIEW

**Comment by davidbenjamin** [comment]
@meganshand This looks good.  Do you want to delete the `LodOddsOverDepthFilter` and remove its threshold from `M2FilteringArgumentCollection` in this PR as well?

**Comment by meganshand** [comment]
I'll defer to @ldgauthier on that question: Should we completely delete the lod divided by depth filter now that the new filtering strategy seems to make it obsolete? I've only run with these changes on the mixtures, so I don't have a good sense if this is the kind of thing we'll need in the future or not?

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/filtering/M2FiltersArgumentCollection.java [hunk 1]
```diff
@@ -137,7 +137,9 @@
     public static final String LONG_INDEL_LENGTH_LONG_NAME = "long-indel-length";
 
     private static final double DEFAULT_LOG_10_PRIOR_OF_SNV = -6.0;
+    private static final double DEFAULT_LOG_10_PRIOR_OF_SNV_FOR_MITO = -2.5;
     private static final double DEFAULT_LOG_10_PRIOR_OF_INDEL = -7.0;
+    private static final double DEFAULT_LOG_10_PRIOR_OF_INDEL_FOR_MITO = -3.75;
```
Comment (ldgauthier) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/filtering/M2FiltersArgumentCollection.java]: These are close-ish to the autosomal priors -- where did they come from?

**Comment by ldgauthier** [comment]
In the latest filtering paradigm, how would somebody who only wanted variants with really high quality bases change the default parameters?

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/filtering/M2FiltersArgumentCollection.java (hunk 1 above)
Comment (meganshand) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/filtering/M2FiltersArgumentCollection.java]: Back of the envelope calculation. I'll add a comment.

**Comment by davidbenjamin** [comment]
>In the latest filtering paradigm, how would somebody who only wanted variants with really high quality bases change the default parameters?

You could decrease `f-score-beta` (default 1.0) to bias the threshold optimization in favor of precision versus sensitivity.

**Comment by ldgauthier** [comment]
> > In the latest filtering paradigm, how would somebody who only wanted variants with really high quality bases change the default parameters?
> 
> You could decrease `f-score-beta` (default 1.0) to bias the threshold optimization in favor of precision versus sensitivity.

Okay.  Rip it out.

**Comment by meganshand** [comment]
@ldgauthier or @davidbenjamin I would appreciate one more quick look now that the lod divided by depth filter has been deleted.

**Comment by davidbenjamin** [comment]
Still :thumbsup: from me.

**Pull request merged**